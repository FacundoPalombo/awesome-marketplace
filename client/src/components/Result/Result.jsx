import PropTypes from "prop-types";
import React from "react";
import styles from "./Result.module";
import { useHistory, Link } from "react-router-dom";

import iconFreeShipping from "assets/icon-shipping.png";
import iconFreeShipping2x from "assets/icon-shipping@2x.png";

export function Result({
  address: { city_name: sellerLocation },
  id,
  price,
  shipping: { free_shipping: freeShipping },
  title,
  thumbnail,
}) {
  return (
    <Link to={`/items/${id}`} className={styles.link}>
      <div className={styles.result}>
        <div className={styles.imageContainer}>
          <img src={thumbnail} className={styles.image} alt={title} />
        </div>
        <div className={styles.description}>
          <h3 className={styles.price}>
            {price}
            {freeShipping && (
              <img
                className={styles.freeShipping}
                srcSet={`${iconFreeShipping} 1x, ${iconFreeShipping2x} 2x`}
              />
            )}
          </h3>
          <h4 className={styles.title}>{title}</h4>
        </div>
        <h4 className={styles.location}>{sellerLocation}</h4>
      </div>
    </Link>
  );
}

Result.propTypes = {
  address: PropTypes.shape({ sellerLocation: PropTypes.string }),
  id: PropTypes.string,
  price: PropTypes.string,
  shipping: PropTypes.shape({ freeShipping: PropTypes.bool }),
  thumbnail: PropTypes.string,
  title: PropTypes.string,
};
