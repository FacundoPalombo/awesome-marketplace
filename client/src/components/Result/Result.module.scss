@import "utils/variables.scss";

.link {
  width: 100%;
  color: $font;
  padding: 16px;
  box-sizing: border-box;
  text-decoration: none;
  &:hover,
  &:visited {
    color: $font;
    outline: none;
  }
  @include md {
    box-sizing: content-box;
    max-height: 140px;
    padding: 0;
  }
}
.result {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: flex-start;
  width: 100%;
  height: 180px;
  @include lg {
    flex-wrap: nowrap;
    box-sizing: content-box;
    height: auto;
  }
}
.image {
  max-width: 180px;
  max-height: 180px;

  margin-left: auto;
  margin-right: auto;
  image-rendering: optimizeSpeed;
  &Container {
    min-width: 180px;
    height: 180px;
    overflow: hidden;
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: $half-unit;
    border: 1px solid #ebebeb99;

    @include md {
      min-width: 140px;
      height: 140px;
    }
  }
  @include md {
    max-width: 140px;
    max-height: 140px;
  }
}
.description {
  font-weight: 400;
  margin-left: $unit-2;
  @include md {
    flex-direction: column-reverse;
  }
}
.price {
  font-size: 24px;
  font-weight: 400;
  @include md {
    margin: $unit 0;
  }
}
.freeShipping {
  margin-left: $half-unit;
}
.title {
  font-size: 18px;
  font-weight: 400;
  margin-right: $unit;
  @include md {
    font-size: 14px;
    width: 100%;
    margin: 0;
  }
}
.location {
  text-align: end;
  font-size: 12px;
  font-weight: 400;
  color: fade-out($color: $font, $amount: 0.3);
  display: block;
  position: relative;
  right: 0;
  margin: 24px 48px auto auto;
  @include lg {
    display: flex;
    position: relative;
    margin: $half-unit $half-unit 0 auto;
  }
}
