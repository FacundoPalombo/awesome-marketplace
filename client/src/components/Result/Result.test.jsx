import React from "react";

import { renderWithRouter } from "../../utils/tests/renderWithRouter";
import { Result } from "./Result";

describe("GIVEN Result Component", () => {
  describe("WHEN results was retrieved", () => {
    it("THEN should match to snapshot", () => {
      const fixtureData = {
        id: "MLA804207795",
        thumbnail:
          "http://http2.mlstatic.com/D_684489-MLA32011256128_082019-O.jpg",
        shipping: { free_shipping: true },
        title: "iPod Touch 32 Gb",
        price: 39999,
        address: { city_name: "Rosario" },
      };
      const { asFragment } = renderWithRouter(<Result {...fixtureData} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
  describe("WHEN results was NOT retrieved", () => {
    it("THEN should match to snapshot", () => {
      const fixtureData = {
        shipping: { free_shipping: null },
        address: { city_name: null },
      };
      const { asFragment } = renderWithRouter(<Result {...fixtureData} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
