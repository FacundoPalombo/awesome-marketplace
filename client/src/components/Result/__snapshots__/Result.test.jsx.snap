// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN Result Component WHEN results was NOT retrieved THEN should match to snapshot 1`] = `
<DocumentFragment>
  <a
    class="link"
    href="/items/:undefined"
  >
    <div
      class="result"
    >
      <div
        class="imageContainer"
      >
        <img
          class="image"
        />
      </div>
      <div
        class="description"
      >
        <h3
          class="price"
        >
          $  
        </h3>
        <h4
          class="title"
        />
      </div>
      <h4
        class="location"
      />
    </div>
  </a>
</DocumentFragment>
`;

exports[`GIVEN Result Component WHEN results was retrieved THEN should match to snapshot 1`] = `
<DocumentFragment>
  <a
    class="link"
    href="/items/:MLA804207795"
  >
    <div
      class="result"
    >
      <div
        class="imageContainer"
      >
        <img
          alt="iPod Touch 32 Gb"
          class="image"
          src="http://http2.mlstatic.com/D_684489-MLA32011256128_082019-O.jpg"
        />
      </div>
      <div
        class="description"
      >
        <h3
          class="price"
        >
          $ 39999 
          <img
            class="freeShipping"
            srcset="file-mock 1x, file-mock 2x"
          />
        </h3>
        <h4
          class="title"
        >
          iPod Touch 32 Gb
        </h4>
      </div>
      <h4
        class="location"
      >
        Rosario
      </h4>
    </div>
  </a>
</DocumentFragment>
`;
